File: Day 13/Longest Palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char,int> cnt;
        int length = 0;
        bool odd = false;
        for (char c : s) cnt[c]++;
        for (auto &p : cnt) {
            length += (p.second / 2) * 2;
            if (p.second % 2) odd = true;
        }
        return odd ? length + 1 : length;
    }
};

File: Day 13/Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        cnt = Counter(s)
        pairs = sum(v // 2 for v in cnt.values())
        return pairs * 2 + (1 if any(v % 2 for v in cnt.values())else 0)
